src/main/perf/SearchPerfTest.java

```diff
@@ -595,4 +613,12 @@ public void run() {
     out.close();
   }
 
+  private static IndexSearcher createIndexSearcher(IndexReader reader, ExecutorService executorService) {
+
+      if (executorService != null) {
```

Comment:
You can just always pass the `executorService` to `IndexSearcher` here -- it interprets `null` properly.